src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py

```diff
@@ -269,23 +268,41 @@ def histogram(
         else:
             data_range = None
 
-        for hist_type, target_object in zip(data_list, target_objects):
-            if len(hist_type) != 0:
-                hist, bins = histogram(hist_type, 20, data_range, density=False)
-            else:
-                hist = empty(0)
-                bins = empty(0)
-            quad = plot.quad(top=hist, bottom=0,
-                             left=bins[:-1], right=bins[1:],
-                             line_color='white', alpha=0.5,
-                             color=color_selector.get_color())
-            legend_manager.add_legend(target_object, quad)
-            hover = HoverTool(
-                tooltips=[(x_label, '@left'), ('The number of samples', '@top')], renderers=[quad]
-                )
-            plot.add_tools(hover)
+        hists = []
+
+        for hist_type in data_list:
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
+            hists.append(hist)
+
+        hists_t = np.array(hists).T
+
+        color_selector = ColorSelectorFactory.create_instance('unique')
+        colors = [color_selector.get_color() for _ in hists_t]
+
+        quad_dicts = {t: [] for t in target_objects}
```

Comment:
There was an annotation omission in my suggestion.
```suggestion
        quad_dicts: dict[HistTypes, list[GlyphRenderer]] = {t: [] for t in target_objects}
```